Mathlib/ModelTheory/Syntax.lean

```diff
@@ -743,6 +742,16 @@ noncomputable def iExs [Finite β] (φ : L.Formula (α ⊕ β)) : L.Formula α :
   let e := Classical.choice (Classical.choose_spec (Finite.exists_equiv_fin β))
   (BoundedFormula.relabel (fun a => Sum.map id e a) φ).exs
 
+variable (β) in
+/-- `iExsUnique f φ` transforms a `L.Formula (α ⊕ β)` into a `L.Formula β` by existentially
+quantifying over all variables `Sum.inr _` and asserting that the solution should be unique  -/
+noncomputable def iExsUnique [Finite β] (φ : L.Formula (α ⊕ β)) : L.Formula α :=
+  let _ := Fintype.ofFinite β
+  iExs β (φ ⊓ iAlls β
+    ((φ.relabel (fun a => Sum.elim (Sum.inl ∘ Sum.inl) Sum.inr a)).imp
+    (BoundedFormula.iInf
+      (fun g => Term.equal (var (Sum.inr g)) (var (Sum.inl (Sum.inr g)))))))
```

Comment:
Do you want to use anonymous dot notation here? Something like
```suggestion
    ((φ.relabel (Sum.elim (.inl ∘ .inl) .inr)).imp
    (.iInf fun g => Term.equal (var (.inr g)) (var (.inl (.inr g))))))
```
and maybe also use `<|` more

Reply:
I've done some of this